src/view/com/post-thread/PostThread.tsx

```diff
@@ -163,42 +167,69 @@ function PostThreadLoaded({
     [threadViewPrefs, thread],
   )
 
-  // construct content
+  const [readyToRender, setReadyToRender] = React.useState(false)
+  const [topPageCount, setTopPageCount] = React.useState(1)
+  const isPrepending = React.useRef(false)
+
   const posts = React.useMemo(() => {
-    let arr = Array.from(
-      flattenThreadSkeleton(
-        sortThread(thread, threadViewPrefs),
-        hasSession,
-        treeView,
-      ),
+    const items = createThreadSkeleton(
+      sortThread(thread, threadViewPrefs),
+      hasSession,
+      treeView,
+      readyToRender,
     )
+
+    // if (!isWeb && !readyToRender) {
+    //   return items.highlightedPost
+    // } else {
+    const postsToPrepend = items.parents?.slice(-(topPageCount * 15)) ?? []
+
+    // Build the entire array of items to render
+    let arr = [
+      // In the case of refreshes we need to take into account the page count when we load
+      ...postsToPrepend,
+      ...items.highlightedPost,
+      ...(items.replies ?? []),
+    ]
+    // Remove any items that shouldn't be visible right now due to view limit
     if (arr.length > maxVisible) {
-      arr = arr.slice(0, maxVisible).concat([LOAD_MORE])
+      arr = [...arr.slice(0, maxVisible), LOAD_MORE]
     }
     if (arr.indexOf(CHILD_SPINNER) === -1) {
       arr.push(BOTTOM_COMPONENT)
     }
+
+    // HACK
+    // This lets us delay rendering of the additional items in the flatlist for a little while so we can latch onto the
+    // correct post
+    if (items.highlightedPost && !readyToRender) {
+      setTimeout(() => {
+        setReadyToRender(true)
+      }, 300)
+    }
+
```

Comment:
This was the best way I could think of to first render the highlight and then render the rest of the posts, but it feels strange. Other ideas?

Reply:
@gaearon I think this is the only remaining quirk to think about. This works perfectly (visually) but curious if there's a better option.